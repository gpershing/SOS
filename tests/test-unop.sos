/*negation of int and float: neg*/
//int
a: int = 5
b: int = -a
print(a) //5
print(b) //-5
print(-1+-2) //-3

//float
c: float = 5.0
d: float = -c
printf(c) //5.0
printf(d) //-5.0
printf(-1.0*-2.0) //2.0


/*negation of bool: not*/
e: bool = True
f: bool = !e
print(e) //True
print(f) //False
print(!!!True) //False
