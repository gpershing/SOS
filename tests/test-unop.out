5
-5
-3
5
-5
2
1
0
0
